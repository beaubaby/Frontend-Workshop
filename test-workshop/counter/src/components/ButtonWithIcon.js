import React from "react";

const Icon = ({ icon }) => <i className={`icon icon-${icon}`} />;
const ButtonWithIcon = ({ icon, children }) => (
  <button>
    <Icon icon={icon} />
    {children}
  </button>
);

export default ButtonWithIcon;
