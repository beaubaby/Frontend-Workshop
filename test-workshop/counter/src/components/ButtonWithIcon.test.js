import React from "react";
import { shallow, render, mount } from "enzyme";
import ButtonWithIcon from "./ButtonWithIcon";

describe("ButtonWithIcon component", () => {
  test("Render ButtonWithIcon component with shollow", () => {
    const wrapper = shallow(
      <ButtonWithIcon icon="coffee">Hey, Jest!</ButtonWithIcon>
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("Render ButtonWithIcon component with static render", () => {
    const wrapper = render(
      <ButtonWithIcon icon="coffee">Hey, Jest!</ButtonWithIcon>
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("Render ButtonWithIcon component with mount", () => {
    const wrapper = mount(
      <ButtonWithIcon icon="coffee">Hey, Jest!</ButtonWithIcon>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
