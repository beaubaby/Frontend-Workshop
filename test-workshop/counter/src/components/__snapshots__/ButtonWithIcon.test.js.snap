// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithIcon component Render ButtonWithIcon component with mount 1`] = `
<ButtonWithIcon
  icon="coffee"
>
  <button>
    <Icon
      icon="coffee"
    >
      <i
        className="icon icon-coffee"
      />
    </Icon>
    Hey, Jest!
  </button>
</ButtonWithIcon>
`;

exports[`ButtonWithIcon component Render ButtonWithIcon component with shollow 1`] = `
<button>
  <Icon
    icon="coffee"
  />
  Hey, Jest!
</button>
`;

exports[`ButtonWithIcon component Render ButtonWithIcon component with static render 1`] = `
<button>
  <i
    class="icon icon-coffee"
  >
    
  </i>
  Hey, Jest!
</button>
`;
